src/main/java/nextstep/qna/domain/Answer.java

```diff
@@ -29,51 +25,45 @@ public Answer(NsUser writer, Question question, String contents) {
     }
 
     public Answer(Long id, NsUser writer, Question question, String contents) {
-        this.id = id;
-        if(writer == null) {
-            throw new UnAuthorizedException();
-        }
-
         if(question == null) {
             throw new NotFoundException();
         }
 
-        this.writer = writer;
+        this.details = new AnswerDetails(id, writer, contents);
         this.question = question;
-        this.contents = contents;
     }
 
     public Long getId() {
-        return id;
-    }
-
-    public Answer setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
+        return details.getId();
     }
 
     public boolean isDeleted() {
         return deleted;
     }
 
     public boolean isOwner(NsUser writer) {
-        return this.writer.equals(writer);
+        return this.details.getWriter().equals(writer);
```

Comment:
현재 객체의 내부 값을 직접 꺼내 동등성 비교를 하고 있네요. 
내부값의 접근제어자가 private일텐데 getter로 꺼내어 직접 연산하면 접근제어자의 의미가 약간 사라질 수 있어요. 
다음과 같이 요청을 하는방식은 어떨까요? 

```suggestion
        return this.details.equalsWriter(writer);
```